awx/ui/src/screens/Inventory/InventorySourceDetail/InventorySourceDetail.js

```diff
@@ -226,16 +275,26 @@ function InventorySourceDetail({ inventorySource }) {
             {t`Edit`}
           </Button>
         )}
-        {user_capabilities?.start && (
-          <InventorySourceSyncButton source={inventorySource} icon={false} />
-        )}
+        {user_capabilities?.start &&
+          (['new', 'running', 'pending', 'waiting'].includes(job?.status) ? (
+            <JobCancelButton
+              job={{ id: job.id, type: 'inventory_update' }}
+              errorTitle={t`Inventory Source Sync Error`}
+              title={t`Cancel Inventory Source Sync`}
+              errorMessage={t`Failed to cancel Inventory Source Sync`}
+              buttonText={t`Cancel Sync`}
+            />
+          ) : (
+            <InventorySourceSyncButton source={inventorySource} icon={false} />
+          ))}
         {user_capabilities?.delete && (
           <DeleteButton
             name={name}
             modalTitle={t`Delete inventory source`}
             onConfirm={handleDelete}
             deleteDetailsRequests={deleteDetailsRequests}
             deleteMessage={t`This inventory source is currently being used by other resources that rely on it. Are you sure you want to delete it?`}
+            isDisabled={job?.status === 'running'}
```

Comment:
I can't remember - can you delete jobs that are in `new`, `pending`, or `waiting`?

Reply:
This change will disable the possibility to delete the inventory source when the sync is running. 

The one to cancel the job itself is here - https://github.com/ansible/awx/blob/edceb0274291f3095a0b2e53bf20bd1256defa0d/awx/ui/src/screens/Inventory/InventorySourceDetail/InventorySourceDetail.js#L279